Owncloud iOs Client/InstantUpload/InstantUpload.m

```diff
@@ -61,30 +61,71 @@ - (void) dealloc {
 }
 
-- (void) setEnabled:(BOOL)enabled {
+- (void) setImageInstantUploadEnabled:(BOOL)enabled {
+    @synchronized (self) {
+        if (enabled != self.imageInstantUploadEnabled) {
+            if (!self.imageInstantUploadEnabled) {
+                if (self.videoInstantUploadEnabled) {
+                    ACTIVE_USER.imageInstantUpload = YES;
+                    [ManageAppSettingsDB updateImageInstantUploadTo:YES];
+                    ACTIVE_USER.timestampInstantUpload = [[NSDate date] timeIntervalSince1970];
+                    [ManageAppSettingsDB updateTimestampInstantUpload:ACTIVE_USER.timestampInstantUpload];
+                    [self attemptUpload];
+                } else {
+                    [self getMediaLibraryPermission:^(BOOL granted) {
+                        if (granted) {
+                            ACTIVE_USER.imageInstantUpload = YES;
+                            [ManageAppSettingsDB updateImageInstantUploadTo:YES];
+                            ACTIVE_USER.timestampInstantUpload = [[NSDate date] timeIntervalSince1970];
+                            [ManageAppSettingsDB updateTimestampInstantUpload:ACTIVE_USER.timestampInstantUpload];
+                            [PHPhotoLibrary.sharedPhotoLibrary registerChangeObserver:self];
+                            [self attemptUpload];
+                        } else {
+                            [self showAlertViewWithTitle:NSLocalizedString(@"access_photos_library_not_enabled", nil) body:NSLocalizedString(@"message_access_photos_not_enabled", nil)];
+                            [self.delegate instantUploadPermissionLostOrDenied];
+                        }
+                    }];
+                }
+            } else{
+                ACTIVE_USER.imageInstantUpload = NO;
+                [ManageAppSettingsDB updateImageInstantUploadTo:NO];
+                if (!self.videoInstantUploadEnabled) {
+                    [PHPhotoLibrary.sharedPhotoLibrary unregisterChangeObserver:self];
+                    [self setBackgroundInstantUploadEnabled:NO];
+                }
+            }
+        }
+    }
+}
+
+- (void) setVideoInstantUploadEnabled:(BOOL)enabled {
     @synchronized (self) {
-        if (enabled != self.enabled) {
-            if (enabled) {
-                if ([PHPhotoLibrary authorizationStatus] == PHAuthorizationStatusAuthorized) {
-                    ACTIVE_USER.instantUpload = YES;
-                    [ManageAppSettingsDB updateInstantUploadTo:YES];
-                    ACTIVE_USER.timestampInstantUpload = [[NSDate date] timeIntervalSince1970];;
+        if (enabled != self.videoInstantUploadEnabled) {
+            if (!self.videoInstantUploadEnabled) {
+                if (self.imageInstantUploadEnabled) {
+                    ACTIVE_USER.videoInstantUpload = YES;
+                    [ManageAppSettingsDB updateVideoInstantUploadTo:YES];
+                    ACTIVE_USER.timestampInstantUpload = [[NSDate date] timeIntervalSince1970];
```

Comment:
You can not use same time stamp for photos and videos, they are going to have different time stamps. If you enabled photos, take some pictures, after a while enabled instant videos, previous photos are not going to be uploaded. Review this